packages/gateway-edge-nextjs/src/cal/cal.ts

```diff
@@ -0,0 +1,155 @@
+import { type NextApiRequest, type NextApiResponse } from "next"
+import ical, { type ICalEventData } from "ical-generator"
+import { type Event } from "@dotkomonline/types"
+import { createServerSideHelpers } from "@trpc/react-query/server"
+import { appRouter, createContextInner, transformer } from "@dotkomonline/gateway-trpc"
+import { getServerSession } from "next-auth"
+import jwt from "jsonwebtoken"
+import { authOptions } from "../../../auth/src/web.app"
+
+const helpers = createServerSideHelpers({
+  router: appRouter,
+  ctx: await createContextInner({
+    auth: null,
+  }),
+  transformer, // optional - adds superjson serialization
+})
+
+function eventUrl(req: NextApiRequest, event: Pick<Event, "id">) {
+  const proto = req.headers["x-forwarded-proto"] || "http"
+  const host = req.headers["x-forwarded-host"] || "online.ntnu.no"
```

Comment:
okay I get that this _works_ in prod/dev/whatever, but this wouldn't work locally? I am fine with merging this without sorting it out, but we should add a config or something for `BASE_URL`/ `HOST`

Reply:
I think it should always work because of nextjs magic, read bottom here (doesn't say it *always* works, but doesn't have a disclaimer either) https://nextjs.org/docs/app/api-reference/functions/headers#ip-address

Reply:
it is set locally for me 🙃

Reply:
i dont see why it would change behavior in prod, everything else being equal?